gengokit/generator/gen.go

```diff
@@ -62,32 +53,28 @@ func GenerateGokit(sd *svcdef.Svcdef, conf gengokit.Config) ([]truss.NamedReadWr
 
 // generateResponseFile contains logic to choose how to render a template file
 // based on path and if that file was generated previously. It accepts a
-// template path to render, a templateExecutor to apply to the template, and a
-// map of paths to files for the previous generation. It returns a
-// truss.NamedReadWriter representing the generated file
-func generateResponseFile(templFP string, te *gengokit.TemplateExecutor, prevGenMap map[string]io.Reader) (truss.NamedReadWriter, error) {
+// template path to render, a executor to apply to the template,
+// and . It returns a truss.NamedReadWriter representing the generated file.
+func generateResponseFile(executor *gengokit.Executor, file io.Reader, templFP string) (truss.NamedReadWriter, error) {
 	var genCode io.Reader
 	var err error
 
 	// Get the actual path to the file rather than the template file path
-	actualFP := templatePathToActual(templFP, te.PackageName)
-	// If we are rendering the server and or the client
-	if templFP == "NAME-service/handlers/server/server_handler.gotemplate" {
-		file := prevGenMap[actualFP]
-		h, err := handler.New(te.Service, file, te.PackageName)
+	actualFP := templatePathToActual(templFP, executor.PackageName)
+
+	switch templFP {
+	case handler.ServerFile:
+		h, err := handler.New(executor.Service, file, executor.PackageName)
 		if err != nil {
 			return nil, errors.Wrapf(err, "cannot parse previous handler: %q", actualFP)
```

Comment:
`file` should probably be named `prevFile`. Then that context should be understandable.